### ISSUE TITLE
fillIn and typeIn ignore maxlength input attribute

### ISSUE THREAD

**Issue by ghost** [emberjs/ember-test-helpers#684]
Both the `fillIn` and `typeIn` helpers appear to ignore the `maxlength` input attribute (although maybe I'm missing something). I am using Ember 3.8. Here's the code I'm testing:

`{{input type="text" class="name-txt" value=name required=true maxlength="10" title=(t "form.message.format.nonBlank")}}`

This works in my app as I would expect: `Earl` is saved without issue, but `Margueritta` is not - a user cannot even enter all characters in the input.

But my below test isn't working. Both names save without issue. I also tried using the `fillIn` helper to no avail. Am I doing something dumb?

```
    assert.expect(1);

    await visit('admin/1');
    await typeIn('.name-txt', 'Margueritta');

    await click('.save-btn');

    assert.equal(this.element.querySelector('input.name-txt').value, "Earl", "did not save; over 10 characters");
```

**Comment by rwjblue** [comment]
Hmm, this definitely seems not great. Would you mind making a failing test case here to make it easier for us to track down?

**Comment by jaydgruber** [comment]
I ran into this same problem.
Not sure it needs to be fixed here, though it is somewhat unexpected behavior.
In my case, I ended up being content with trusting the browser to fulfill its contract, and went with:
```javascript
assert.dom(selector).hasAttribute('maxlength', maxLength);
```
Went ahead and added test cases though, since it was a quick ask. #747

**Comment by hjdivad** [comment]
fixed in #747
